pkg/gadgets/internal/socketenricher/tracer.go

```diff
@@ -162,10 +166,55 @@ func (se *SocketEnricher) start() error {
 		return fmt.Errorf("read BPF iterator: %w", err)
 	}
 
+	cleanupIter, err := link.AttachIter(link.IterOptions{
+		Program: se.objs.IgSkCleanup,
+		Map:     se.objs.Sockets,
+	})
+	if err != nil {
+		return fmt.Errorf("attach BPF iterator for cleanups: %w", err)
+	}
+	se.links = append(se.links, cleanupIter)
+
+	se.done = make(chan bool)
+	go se.cleanupDeletedSockets(cleanupIter)
+
+	return nil
+}
+
+func (se *SocketEnricher) cleanupDeletedSockets(cleanupIter *link.Iter) {
+	ticker := time.NewTicker(2 * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-se.done:
+			return
+		case <-ticker.C:
+			err := se.cleanupDeletedSocketsNow(cleanupIter)
+			if err != nil {
+				fmt.Printf("socket enricher: %v\n", err)
+			}
+		}
+	}
+}
+
+func (se *SocketEnricher) cleanupDeletedSocketsNow(cleanupIter *link.Iter) error {
+	file, err := cleanupIter.Open()
+	if err != nil {
+		return fmt.Errorf("open BPF iterator: %w", err)
+	}
+	defer file.Close()
+	_, err = io.ReadAll(file)
+	if err != nil {
+		return fmt.Errorf("read BPF iterator: %w", err)
+	}
 	return nil
```

Comment:
offtopic: I'm wondering if cilium/ebpf should provide an easier way to run iterators. Something like `cleanupIter.Run()` would be nice for cases the iter doesn't generate any output.